### PR TITLE
MATIC support for hardcoded liquidity pool addresses

### PR DESCRIPTION
**Pull request by schystz**
## Description of Changes

Liquidity pools on the frontend app is currently hardcoded. This PR adds support to a new hardcoded list (`REACT_APP_LIQUIDITY_POOL_ADDRESSES_MATIC`) which is loaded instead of the default list if network is MATIC.

**IMPORTANT: Please note that currently added matic pool addresses are production pools and meant to be used for app.halodao.com so please refrain from adding/removing liquidity unless well coordinated with the team**

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [ ] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [x] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

**Comment by chris247474** [comment]
image

@schystz Good work getting this up!! Noticed small UI things, just initial look through:
1. `RNBW Supply` label on Polygon looks like its reflecting the total wrapped tokens and not RNBW supply on mainnet - how would you prefer to handle this? I guess either we can (a) remove the `RNBW Supply` label on Polygon OR (b) fetch the total supply from mainnet via either coingecko or directly by calling the `RNBW.totalSupply()` function
2. The upper right RNBW balance shows as 0 when on Polygon network

@schystz @tracyarciaga do you prefer that I make separate bug tickets for this so this PR can move along? Or address in here?

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Comment by tracyarciaga** [comment]
> image
> 
> @schystz Good work getting this up!! Noticed small UI things, just initial look through:
> 
> 1. `RNBW Supply` label on Polygon looks like its reflecting the total wrapped tokens and not RNBW supply on mainnet - how would you prefer to handle this? I guess either we can (a) remove the `RNBW Supply` label on Polygon OR (b) fetch the total supply from mainnet via either coingecko or directly by calling the `RNBW.totalSupply()` function
> 2. The upper right RNBW balance shows as 0 when on Polygon network
> 
> @schystz @tracyarciaga do you prefer that I make separate bug tickets for this so this PR can move along? Or address in here?

This can be logged as a separate bug ticket @schystz @ctverceles

**Comment by tracyarciaga** [comment]
Adding the two additional issues from @ctverceles here:
1. Invalid address error when loading pools 

https://user-images.githubusercontent.com/6169194/132787043-c76e2ce5-a23c-45ec-ba53-55240cf8e5de.mov

2. Polygon pools not refreshing switching the network back to Polygon


https://user-images.githubusercontent.com/6169194/132789463-a935272d-bf45-491b-bda0-abfb39244e67.mov

File: src/halo-hooks/amm/useLiquidityPool.ts [hunk 1]
```diff
@@ -8,15 +8,14 @@ import { formatEther } from 'ethers/lib/utils'
 import { ERC20_ABI } from 'constants/abis/erc20'
 import { getContract } from 'utils'
 import { Token } from '@sushiswap/sdk'
-import { HALO_REWARDS_ADDRESS } from '../../constants'
+import { HALO_REWARDS_V1_ADDRESS } from '../../constants'
```
Comment (tracyarciaga) [inline src/halo-hooks/amm/useLiquidityPool.ts:11]: @schystz FYI, I changed this to V1 to fix the invalid address issue.

**Review by tracyarciaga — APPROVED** [review]
Additional issues have been resolved and approving this now.

File: src/halo-hooks/amm/useLiquidityPool.ts (hunk 1 above)
Comment (schystz) [reply src/halo-hooks/amm/useLiquidityPool.ts:11]: Oh thanks for the catch @tracyarciaga ! Looks good to me

**Pull request merged**